packages/qwik-city/src/middleware/request-handler/request-event.ts

```diff
@@ -220,6 +221,19 @@ export function createRequestEvent(
       return new RedirectMessage();
     },
 
+    rewrite: (url: string) => {
```

Comment:
Incidentally, I'd also like us to support location objects for easy parameter changing.